Update ERC-7066: Move to Last Call (#47)
ethereum/ERCs: ERCS/erc-7066.md

description: Interface for enabling locking of ERC-721 using locker and approved
author: Piyush Chittara (@piyush-chittara), StreamNFT (@streamnft-tech), Srinivas Joshi (@SrinivasJoshi)
discussions-to: https://ethereum-magicians.org/t/eip-7066-lockable-extension-for-erc721/14425

> status: Final

Reviewer: You'll need to begin a Last Call period and add a `last-call-deadline` before you can go to Final. The Last Call status gives any potential reviewers notice that the proposal is about to go to Final.

 ```suggestion
status: Last Call
last-call-deadline: 2023-11-20
```

## Abstract

> An extension of [ERC-721](./erc-721.md), this standard incorporates `locking` features into NFTs, allowing for various uses while preventing sale or transfer. The token's `owner` can `lock` it, setting up locker address (either an EOA or a contract) that exclusively holds the power to unlock the token. Owner can also provide approval for tokenId, enabling ability to lock asset while address holds the token approval. Token can also be locked by `approved`, assigning locker to itself. Upon token transfer, these rights get purged.

Reviewer: ```suggestion
An extension of [ERC-721](./erc-721.md), this standard incorporates `locking` features into NFTs, allowing for various uses while preventing sale or transfer. The token's `owner` can `lock` it, setting up locker address (either an EOA or a contract) that exclusively holds the power to unlock the token. Owner can also provide approval for `tokenId`, enabling ability to lock asset while address holds the token approval. Token can also be locked by `approved`, assigning locker to itself. Upon token transfer, these rights get purged.
```

## Motivation

> [ERC-721](./erc-721.md) has sparked an unprecedented surge in demand for NFTs. However, despite this tremendous success, NFT economy suffers from secondary liquidity where it remains Illiquid in owner’s wallet. There are projects such as NFTfi, Paraspace which aims to address the liquidity challenge, but they entail below mentioned inconveniences and risks for owners as they necessitate transferring the participating NFTs to the projects' contracts. 

Reviewer: Should probably avoid mentioning specific projects:

```suggestion
[ERC-721](./erc-721.md) has sparked an unprecedented surge in demand for NFTs. However, despite this tremendous success, the NFT economy suffers from secondary liquidity where it remains illiquid in owner’s wallet. There are projects which aim to address the liquidity challenge, but they entail the below mentioned inconveniences and risks for owners as they necessitate transferring the participating NFTs to the projects' contracts. 
```

- Loss of utility: The utility value of NFTs diminishes when they are transferred to an escrow account, no longer remaining under the direct custody of the owners.
- Lack of composability: The market could benefit from increased liquidity if NFT owners had access to multiple financial tools, such as leveraging loans and renting out their assets for maximum returns. Composability serves as the missing piece in creating a more efficient market.
- Smart contract vulnerabilities: NFTs are susceptible to loss or theft due to potential bugs or vulnerabilities present in the smart contracts they rely on.

The aforementioned issues contribute to a poor user experience (UX), and we propose enhancing the [ERC-721](./erc-721.md) standard by implementing a native locking mechanism: 
Rather than being transferred to a smart contract, an NFT remains securely stored in self-custody but is locked. 
During the lock period, the NFT's transfer is restricted while its other properties remain unchanged. 

> NFT Owner retains the ability to use or distribute it’s utility

Reviewer: ```suggestion
NFT Owner retains the ability to use or distribute it’s utility.
```

> - This implementation proposes a separation of rights between locker and approver. Token can be locked with approval and approved can unlock and withdraw tokens (opening up opportunities like renting, lending, bnpl etc), and token can be locked lacking the rights to revoke token, yet can unlock if required (opening up opportunities like account-bound NFTs).

Reviewer: ```suggestion
- This implementation proposes a separation of rights between locker and approver. Token can be locked with approval and approved can unlock and withdraw tokens (opening up opportunities like renting, lending, BNPL etc), and token can be locked lacking the rights to revoke token, yet can unlock if required (opening up opportunities like account-bound NFTs).
```

Also, please include the abbreviation on line 38:

> - Buy Now Pay Later (BNPL): ...

We prefer that abbreviations are expanded the first time they are used.
